### PR TITLE
End of an era - Remove Juju 1 support

### PR DESCRIPTION
**Pull request by hatched**
To speed up development of new features this branch removes support for Juju 1 from the GUI. Juju 1 is still supported in the GUI up to the 2.4.4 release and any critical updates will be forked from there and applied.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6980/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6981/

**Comment by huwshimi** [comment]
QA seems OK with guiproxy.

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js [hunk 1]
```diff
@@ -629,7 +626,7 @@ YUI.add('deployment-flow', function() {
               </juju.components.USSOLoginLink>
             </div>
             <div className="deployment-login__signup">
-              Don't have an account?
+              Do not have an account?
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: Anti-contraction?

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js [hunk 2]
```diff
@@ -30,14 +30,12 @@ YUI.add('charmbrowser-component', function() {
       charmstoreSearch: React.PropTypes.func.isRequired,
       charmstoreURL: React.PropTypes.string.isRequired,
       deployService: React.PropTypes.func.isRequired,
-      displayPlans: React.PropTypes.bool.isRequired,
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: Why is displayPlans being removed?

File: jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js [hunk 3]
```diff
@@ -54,8 +54,6 @@ YUI.add('service-overview', function() {
       const props = this.props;
 
       if (!props.displayPlans) {
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js]: Why remove displayPlans elsewhere but not here?

File: jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js [hunk 4]
```diff
@@ -237,8 +235,7 @@ YUI.add('service-overview', function() {
                 id: serviceId,
                 activeComponent: 'change-version'}}}});
       }
-      // If we aren't in a Juju 2 model then do not query for
-      // or display the plans.
+
       if (this.props.displayPlans && (state.activePlan || plans)) {
```
Comment (kadams54) [inline jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js]: Or here?

File: jujugui/static/gui/src/test/test_app.js [hunk 5]
```diff
@@ -798,51 +736,6 @@ describe('App', function() {
         'navigate should not be called in gisf mode here');
     });
 
-    // XXX This test causes intermittent cascading failures when run in CI.
-    // When the notification system gets refactored this test can be un-skipped.
```
Comment (kadams54) [inline jujugui/static/gui/src/test/test_app.js]: Removing this makes me cry with joy.

File: jujugui/static/gui/src/test/test_app.js [hunk 6]
```diff
@@ -2518,41 +2287,6 @@ describe('App', function() {
     });
   });
 
-  describe('_sharingVisibility', function() {
```
Comment (kadams54) [inline jujugui/static/gui/src/test/test_app.js]: Hah, whoops! I'm pretty sure I'm to blame for this useless/empty test suite.

File: jujugui/static/gui/src/test/test_bundle_import_notifications.js [hunk 7]
```diff
@@ -124,168 +124,6 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
     });
   });
 
-  describe('bundle helpers watchAll', function() {
```
Comment (kadams54) [inline jujugui/static/gui/src/test/test_bundle_import_notifications.js]: I feel like I'm missing some piece of juju knowledge here; how is this connected to legacy juju?

File: jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js]: It made my editors and github throw a styling fit :)

File: jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js]: displayPlans was only toggled on legacy Juju - which means that it has been `true` all the time in Juju 2

File: jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js (hunk 3 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js]: It was basically an inverse of the above comment. It is unused so always defaults to false in Juju 2

File: jujugui/static/gui/src/test/test_app.js (hunk 6 above)
Comment (hatched) [inline jujugui/static/gui/src/test/test_app.js]: haha

File: jujugui/static/gui/src/test/test_bundle_import_notifications.js (hunk 7 above)
Comment (hatched) [inline jujugui/static/gui/src/test/test_bundle_import_notifications.js]: Back in the olden times we used to get deltas on bundle imports - they weren't just individual api calls.

File: jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js (hunk 3 above)
Comment (kadams54) [inline jujugui/static/gui/src/app/components/inspector/service-overview/service-overview.js]: Still not sure why the if statement and prop remain. If it's unused/always false why not nuke it?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/6988/

**Comment by hatched** [comment]
Thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**